qiita_db/test/test_metadata_template.py

```diff
@@ -652,6 +775,133 @@ def test_create(self):
                ['2.Sample3', "Value for sample 3"]]
         self.assertEqual(obs, exp)
 
+    def test_create_int_prefix(self):
+        """Creates a new SampleTemplate"""
+        st = SampleTemplate.create(self.metadata_int_pref, self.new_study)
+        # The returned object has the correct id
+        self.assertEqual(st.id, 2)
+
+        # The relevant rows to required_sample_info have been added.
+        obs = self.conn_handler.execute_fetchall(
+            "SELECT * FROM qiita.required_sample_info WHERE study_id=2")
+        # sample_id study_id physical_location has_physical_specimen
+        # has_extracted_data sample_type required_sample_info_status_id
+        # collection_timestamp host_subject_id description
+        exp = [["2.12.Sample1", 2, "location1", True, True, "type1", 1,
```

Comment:
I would say that is not that ambiguous. The first dot separates the
study_id from whatever the user used as a sample id. An the second part is
completely arbitrary...

2015-02-11 13:22 GMT-08:00 Daniel McDonald <notifications@github.com>:

> In qiita_db/test/test_metadata_template.py
> <https://github.com/biocore/qiita/pull/843#discussion_r24534521>:
>
> > @@ -652,6 +775,133 @@ def test_create(self):
> >                 ['2.Sample3', "Value for sample 3"]]
> >          self.assertEqual(obs, exp)
> >
> > +    def test_create_int_prefix(self):
> > +        """Creates a new SampleTemplate"""
> > +        st = SampleTemplate.create(self.metadata_int_pref, self.new_study)
> > +        # The returned object has the correct id
> > +        self.assertEqual(st.id, 2)
> > +
> > +        # The relevant rows to required_sample_info have been added.
> > +        obs = self.conn_handler.execute_fetchall(
> > +            "SELECT * FROM qiita.required_sample_info WHERE study_id=2")
> > +        # sample_id study_id physical_location has_physical_specimen
> > +        # has_extracted_data sample_type required_sample_info_status_id
> > +        # collection_timestamp host_subject_id description
> > +        exp = [["2.12.Sample1", 2, "location1", True, True, "type1", 1,
>
> The issue is the ambiguity, and that we are perpetuating the ambiguity. I
> don't think there is a great solution here, but we *may* a more unique
> delimiter would reduce the potential for confusion
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/biocore/qiita/pull/843/files#r24534521>.
>



-- 
Jose Navas